src/rust/engine/src/nodes.rs

```diff
@@ -1205,13 +1209,23 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let span_id = generate_random_64bit_string();
-    let node_workunit_params = if context.session.should_record_zipkin_spans() {
-      let node_name = format!("{}", self);
-      let start_time = std::time::SystemTime::now();
-      Some((node_name, start_time, span_id.clone()))
-    } else {
-      None
+
+    let handle_workunits =
```

Comment:
If the effect is that these act identically here, I think that it would be fine to merge them?

Since so many fewer `@rules` are likely to be recorded than before, I think that it should be safe from a performance perspective... thoughts @cattibrie , @pierrechevalier83 , @illicitonion ?

Reply:
For posterity, this ended up as a followup in #8658.